### ISSUE TITLE
[CLI]: wandb.log(..., commit = False) doesnt work and continues to log metrics to the server

### ISSUE THREAD

**Issue by yiphei** [wandb/wandb#7013]
### Describe the bug

Im trying to use `wandb.log(..., commit = False)` because I do a lot of spot instance training, so when I reload the checkpoint, I need to make sure that my wandb resumes at the right step. Because I don't save checkpoints at every step, the flag only becomes True (e.g. `wandb.log(..., commit = True)` ) when I save the checkpoint. However, somehow `wandb.log(..., commit = False)` just doesnt work and continue to log to the server. This is my current implementation:

```python
# I oversimplified my code and omitted details just for illustrative purposes
while iter_num < TRAIN_CONFIG.TRAIN_STEPS:
    if iterm_num % est_interval == 0:
        ...
        wandb.log(
            {
                "est_train_loss": train_loss,
                "est_val_loss": val_loss,
                "est_lr": lr,
            },
            step = iter_num,
            commit = True,
        )
        ...
        torch.save(
            checkpoint,
            (
                f"transformer_dropout/model_checkpoints/ckpt.pt"
                if args.is_local
                else os.path.join(checkpoint_path, "ckpt.pt")
            ),
        )
    ...
    logits, loss = model(X,Y)
    ...

    wandb.log(
        {
            "loss": running_loss,
            "dropout_entropy": running_entropy,
            "dropout_l1_norm": running_l1_norm,
            "time": float(f"{dt*1000:.2f}"),
        },
        step = iter_num,
        commit = False
    )
    ...
```
Im also open to other ways of ensuring continuity in my wandb run when I resume a training run from a checkpoint

### Environment

WandB version: 0.16.3

OS: 14.3.1 (23D60)

Python version: 3.10.1

Versions of relevant libraries:

**Comment by ArtsiomWB** [comment]
Hi @yiphei! Thank you for writing in!

To confirm, when you are adding commit = False, you are interested in seeing the metrics not get uploaded to the wandb.ai, but it gets recorded regardless of you having the flag?

**Comment by yiphei** [comment]
@ArtsiomWB correct! `commit=False` is just a no-op for me

**Comment by yiphei** [comment]
@ArtsiomWB gentle bump

**Comment by yiphei** [comment]
@kptkin, @ArtsiomWB gentle bump

**Comment by yiphei** [comment]
@kptkin @ArtsiomWB bump

**Comment by mvmacfarlane** [comment]
@kptkin @ArtsiomWB bump

**Comment by kptkin** [comment]
@mvmacfarlane could you please provide more details on your issue? 
i provided the explanation of the current behavior with `commit=False` as well as an option to use `define_metric` which will avoid the need to pass a step to wandb completely and allow you to define your own custom x-axis.

if none of these help with your case a more detailed explanation of your specific use-case would be helpful to help us figure out how to best assist you.

**Comment by sydholl** [comment]
Due to inactivity we are closing this request, please comment to reopen.
